core/pkg/fixhandler/fixhandler.go

```diff
@@ -185,9 +186,22 @@ func (h *FixHandler) PrintExpectedChanges(resourcesToFix []ResourceFixInfo) {
 func (h *FixHandler) ApplyChanges(resourcesToFix []ResourceFixInfo) (int, []error) {
 	updatedFiles := make(map[string]bool)
 	errors := make([]error, 0)
+	// Map with key as filepath
+	filePathFixInfo := make(map[string]*FileFixInfo)
 	for _, resourceToFix := range resourcesToFix {
 		singleExpression := reduceYamlExpressions(&resourceToFix)
-		if err := h.applyFixToFile(resourceToFix.FilePath, singleExpression); err != nil {
+		resourceFilePath := resourceToFix.FilePath
+
+		if _, pathExistsInMap := filePathFixInfo[resourceFilePath]; !pathExistsInMap {
+			contentToAdd := make([]ContentToAdd, 0)
+			linesToRemove := make([]LinesToRemove, 0)
+			filePathFixInfo[resourceFilePath] = &FileFixInfo{
+				ContentToAdd:  contentToAdd,
+				LinesToRemove: linesToRemove,
+			}
+		}
+
+		if err := h.updateFileFixInfo(resourceFilePath, singleExpression, resourceToFix.DocumentIndex, filePathFixInfo[resourceFilePath]); err != nil {
```

Comment:
* Applying fixes to files is expensive. So, the fewer times we do it, the better. If we iterate over resources and apply fixes we might have to fix the file multiple times because there could be multiple resources in a single file.

Reply:
* Regarding the question about `filePathFixInfo`, as I stated before, multiple resources can be a part of the same YAML file. `filePathFixInfo` is used to organize all the fixes required for the file in a single place. 

* Building `filepathFixInfo` became complicated because `A single filepath can have different resources`.